File: 1212-sequential-digits/sequential-digits.py
class Solution:
    def sequentialDigits(self, low: int, high: int) -> List[int]:
        res = []
        def rec(num):
            if low <= num <= high:
                res.append(num)
            elif num > high: return
            val = num % 10
            if val == 9: return
            num = (num * 10) + (val + 1) 
            # print(val, num)
            rec(num)
        
        for i in range(1, 9):
            rec(i)
        return sorted(res)
        

File: 1212-sequential-digits/sequential-digits.java
class Solution {
    public void findSeq(int low, int high, int num, List<Integer> res){
        if(num >= low && num <= high){
            res.add(num);
        }
        else if(num > high) return;
        int val = num % 10;
        if(val == 9) return;
        num = (num * 10) + (val + 1);
        findSeq(low, high, num, res);
    }
    public List<Integer> sequentialDigits(int low, int high) {
        List<Integer> res = new ArrayList<>();
        for(int i = 1; i < 9; i++){
            findSeq(low, high, i, res);
        }
        Collections.sort(res);
        return res;
    }
}